### PR TITLE
feat: Cleaning out obsolete challenge records, fixing certificate save

### LINKED ISSUE #248: Create queue and worker for ACME challenge DNS cleanup
It does not matter if the cert creation process completed or failed, we need to make sure to remove the challenge records from the DNS.

### PR DESCRIPTION
**Pull request by dadolhay**
Closes #248

### CODE REVIEW

File: app/queues/certificate/order-completer-worker.server.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const orderCompleterWorker = new Worker<CertificateJobData>(
       throw e;
     }
 
-    certificateModel.updateCertificateById(certificateId, {
+    await certificateModel.updateCertificateById(certificateId, {
```
Comment (humphd) [inline app/queues/certificate/order-completer-worker.server.ts:76]: Ah, we missed this!  Good find.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -33,6 +33,7 @@ model DnsRecord {
   expiresAt    DateTime
   lastNotified DateTime?
   user         User            @relation(fields: [username], references: [username], onDelete: Cascade)
+  challenge    Challenge[]
```
Comment (humphd) [inline prisma/schema.prisma:36]: This feels off to me.  Can you explain this?  Is there no where else we can store this relation?  It feels odd for a record to reference other records.  Even putting on the User would feel a bit better.

Comment (dadolhay) [reply prisma/schema.prisma:36]: What do you think?

Comment (humphd) [reply prisma/schema.prisma:36]: That makes sense to me.

Comment (dadolhay) [reply prisma/schema.prisma:36]: Okay, I'll refactor

**Review by humphd — APPROVED** [review]

**Comment by humphd** [comment]
@cychu42 can you review this please?

**Comment by Eakam1007** [comment]
Is this related to an issue? What is the context here?

File: app/queues/certificate/dns-cleaner-worker.server.ts [hunk 3]
```diff
@@ -9,8 +11,24 @@ export const dnsCleanerQueueName = 'certificate-cleanDns';
...
+     * the database relation is set to cascade - challenge delete
+     * will remove the row from the dnsRecords table as well
+     *
+     * After those records were cascade deleted, we can trigger
+     * the reconciler as well
+     */
+
+    await setIsReconciliationNeeded(true);
+
+    logger.info('Challenges removed (cascading to records), reconciler triggered');
```
Comment (Eakam1007) [inline app/queues/certificate/dns-cleaner-worker.server.ts:31]:
```suggestion
    logger.info('Challenges removed (cascading to dns records), reconciler triggered');
```
Slightly more consistent with rest of the code

**Comment by dadolhay** [comment]
@Eakam1007 
Issue attached to PR

This is the last step in the cert creation process, removing the outdated DNS records

**Review by Eakam1007 — APPROVED** [review]
One small suggestion but look good

**Review by Eakam1007 — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**